modules/Kconfig

```diff
@@ -31,7 +31,6 @@ source "modules/Kconfig.nxp_s32"
 source "modules/Kconfig.silabs"
 source "modules/Kconfig.simplelink"
 source "modules/Kconfig.sof"
-source "modules/Kconfig.st"
 source "modules/Kconfig.stm32"
```

Comment:
> Genuine question: No need to add `modules/hal_st/Kconfig` here ?

had the same question myself, but it seems that others (like hal_nxp) don't have it.

Reply:
looks like the build systems adds that for you in `build/Kconfig/Kconfig.modules`

Reply:
...not sure how it gets included when the module is not checked out though, but it seems like it does.

Reply:
oh, this, https://github.com/zephyrproject-rtos/zephyr/blob/main/modules/Kconfig#L118-L123, yeah so if the hal_st module is there it gets included from the generated file and the symbols are usable, otherwise it gets included by the wildcard with all symbols excluded, which is exactly the same because all symbols are just placeholders but I guess it's the right thing to do anyway

Reply:
> oh, this, https://github.com/zephyrproject-rtos/zephyr/blob/main/modules/Kconfig#L118-L123, yeah so if the hal_st module is there it gets included from the generated file and the symbols are usable, otherwise it gets included by the wildcard with all symbols excluded, which is exactly the same because all symbols are just placeholders but I guess it's the right thing to do anyway

Thanks for explanation!